exp/views/study.py

```diff
@@ -434,12 +444,30 @@ def get_initial(self):
             initial['structure'] = json.dumps(structure)
         return initial
 
+    def post(self, request, *args, **kwargs):
+        """
+        Form for allowing user to change study type and study metadata fields
+        """
+        study = self.get_object()
+        if 'structure' not in self.request.POST:
+            study.metadata = self.extract_type_metadata()
+            study.study_type = StudyType.objects.get(id=self.request.POST.get('study_type'))
```

Comment:
You could do:
```
study.study_type_id = self.request.POST.get('study_type')
```
Assuming that you're passing the PK as study_type in the POST.

Reply:
Your way does validate that the study type exists tho. If you still want to do that you could `StudyType.objects.filters(id=self.request.POST.get('study_type')).values_list('id', flat=True)[0]`

Reply:
i don't understand why you want to set study_type_id instead of study_type. does this save time?

Reply:
`study.study_type` is really just Django ORM magic, in the database there's a field on `studies_study` table called `study_type_id` which is the ForeignKey to the `studies_studytype` table. You can bypass instantiating the model, doing the query to get it: `StudyType.objects.get(id=self.request.POST.get('study_type'))` and just set `study.study_type_id` to the PK of the StudyType and it will work. Make sense?

Reply:
yes, that's fancy :)